### PR TITLE
feat: refactor metering ,support Third-party resource access

### PR DESCRIPTION
**Pull request by xiao-jay**
Signed-off-by: 晓杰 <2561589453@qq.com>
[架构描述](https://github.com/xiao-jay/sealos/tree/metering-dev/controllers/metering#metering%E7%BB%93%E6%9E%84%E5%9B%BE)

- [x] 代码开发完成
- [x] 列好测试流程
- [x] 上线完成
- [x] 进行e2e测试
- [x] podcontroller 里面storage还无法计费
- [x] 编写如何进行e2e测试文档
- [x] 完成第三方controller如何接入文档
- [x] 修改metering设计文档， 写成proposal
- [ ] 让用户不能get到metering-quota

### CODE REVIEW

File: controllers/cluster/.gitignore [hunk 1]
```diff
@@ -23,3 +23,4 @@ testbin/*
 *.swp
 *.swo
 *~
+*DS_Store
```
Comment (zzjin) [inline controllers/cluster/.gitignore:26]: 统一都改成**/.DS_Store好点

Comment (xiao-jay) [reply controllers/cluster/.gitignore:26]: ok

File: controllers/metering/api/v1/deduction_types.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+// DeductionStatus defines the observed state of Deduction
+type DeductionStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Deduction is the Schema for the deductions API
+type Deduction struct {
```
Comment (fanux) [inline controllers/metering/api/v1/deduction_types.go:38]: Not a noun, should not use an action as CR type

File: controllers/metering/api/v1/extensionresourcesprice_types.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ExtensionResourcesPriceSpec   `json:"spec,omitempty"`
+	Status ExtensionResourcesPriceStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// ExtensionResourcesPriceList contains a list of ExtensionResourcesPrice
+type ExtensionResourcesPriceList struct {
```
Comment (fanux) [inline controllers/metering/api/v1/extensionresourcesprice_types.go:63]: Don't need this list any more.

File: controllers/metering/api/v1/meteringnamespace_types.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+// MeteringNamespaceStatus defines the observed state of MeteringNamespace
+type MeteringNamespaceStatus struct {
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// MeteringNamespace   is the Schema for the meteringnamespaces API
+type MeteringNamespace struct {
```
Comment (fanux) [inline controllers/metering/api/v1/meteringnamespace_types.go:35]: Why need this CR?

**Comment by zzjin** [comment]
> Signed-off-by: 晓杰 [2561589453@qq.com](mailto:2561589453@qq.com) [架构描述](https://github.com/xiao-jay/sealos/tree/metering-dev/controllers/metering#metering%E7%BB%93%E6%9E%84%E5%9B%BE)
> 
> * [x]  代码开发完成
> * [x]  列好测试流程
> * [x]  上线完成
> * [x]  进行e2e测试
> * [x]  podcontroller 里面storage还无法计费
> * [x]  编写如何进行e2e测试文档
> * [x]  完成第三方controller如何接入文档
> * [x]  修改metering设计文档， 写成proposal
> * [ ]  让用户不能get到metering-quota

This pr seems too big to review, maybe we can split into some small chunks, like:
* one pr for e2e tests
* one pr for typos (gitignore)
* one pr for docs
* one pr for porposal

**Pull request closed without merging**